src/vm/dllimport.cpp

```diff
@@ -6096,7 +6104,42 @@ static void DetermineLibNameVariations(const char* const** libNameVariations, in
         *numberOfVariations = COUNTOF(NameOnly);
     }
 }
-#endif
+#else // FEATURE_PAL
+static void DetermineLibNameVariations(const char* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+{
+    bool endWithDot = libName.EndsWith(W(".")) == TRUE;
```

Comment:
It is a bad practice to compare `BOOL` to `TRUE`. `BOOL` is actually integer and any non-zero value can mean true. `TRUE` is defined as `1`. So comparing `BOOL` to `TRUE` assumes that the non-zero true value is `1` that is not always the case.

Use either `!!libName.EndsWith(W("."))`  or `libName.EndsWith(W(".")) != FALSE`.

Reply:
Thanks for this. Previous I had searched the implementation for SString.EndWith and it always return TRUE or FALSE.  I will use !! instead.